### ISSUE TITLE
Pallas YAT attention computes incorrect dQ/dK gradients across multiple K/V tiles

### ISSUE THREAD

**Issue by mlnomadpy** [azettaai/nmn#48]
## Summary

The custom VJP in `pallas_yat_l1_attention` produces materially incorrect gradients for Q and K whenever attention spans more than one K/V tile. dV remains correct in the same reproduction.

## Affected code

- `src/nmn/nnx/layers/attention/pallas_yat_attention.py`
- `_yat_l1_bwd_dkv_kernel` and `_yat_l1_bwd_dq_kernel`

Each backward tile computes the normalization correction `sum(dW * W)` using only the current K/V tile. For L1-normalized attention, that correction must be accumulated across the full key sequence.

## Reproduction

On commit `4f0298d`, using JAX CPU Pallas interpret mode:

```python
import jax
import jax.numpy as jnp
from nmn.nnx.layers.attention.pallas_yat_attention import pallas_yat_l1_attention
from nmn.nnx.layers.attention.fused_yat_attention import fused_yat_l1_attention

q = jax.random.normal(jax.random.key(3), (1, 8, 1, 4))
k = jax.random.normal(jax.random.key(4), (1, 8, 1, 4))
v = jax.random.normal(jax.random.key(5), (1, 8, 1, 4))

def p_loss(q, k, v):
    return jnp.sum(pallas_yat_l1_attention(
        q, k, v, block_q=4, block_k=4, interpret=True
    ) ** 2)

def ref_loss(q, k, v):
    return jnp.sum(fused_yat_l1_attention(q, k, v) ** 2)

gp = jax.grad(p_loss, argnums=(0, 1, 2))(q, k, v)
gr = jax.grad(ref_loss, argnums=(0, 1, 2))(q, k, v)
print([float(jnp.max(jnp.abs(a - b))) for a, b in zip(gp, gr)])
```

Observed maximum errors:

```
dQ: 3.73876
dK: 3.93280
dV: 2.38e-7
```

Relative max errors are about 60% for dQ and 49% for dK.

The current gradient tests use a large tensor with a mean-reduced loss and an absolute tolerance of `5e-3`, which scales the wrong gradients enough for the test to pass.

## Impact

Training with the Pallas path can update Q/K projections using incorrect gradients on GPU/TPU even when forward outputs match the reference.

## Acceptance criteria

- [ ] dQ, dK, and dV match the fused reference for one and multiple K/V tiles.
- [ ] Tests include scale-independent relative/cosine checks or a sum-reduced loss that exposes this regression.
- [ ] Causal and non-causal backward parity are covered.
- [ ] The corrected kernel is validated in `interpret=True` and on a supported GPU or TPU backend.

**Comment by mlnomadpy** [comment]
PR #83 fixed the multi-tile gradient semantics and passed independent CPU Pallas `interpret=True` parity review, but the issue acceptance criterion also requires execution on a supported GPU or TPU backend. That hardware validation has not been run locally, so this issue is reopened until accelerator lowering/execution is verified.
